packages/api/models/executions.js

```diff
@@ -136,36 +105,54 @@ class Execution extends Manager {
     return Promise.all(updatePromises);
   }
 
-  /**
-   * Update an existing execution record, replacing all fields except originalPayload
-   * adding the existing payload to the finalPayload database field
-   *
-   * @param {Object} message - A workflow execution message
-   * @returns {Promise<Object>} An execution record
-   */
-  async updateExecutionFromSns(message) {
-    const record = await Execution.generateRecord(message);
-    return this.create(record);
-  }
-
-  /**
-   * Create a new execution record from incoming SNS messages
-   *
-   * @param {Object} message - A workflow execution message
-   * @returns {Promise<Object>} An execution record
-   */
-  async createExecutionFromSns(message) {
-    const record = await Execution.generateRecord(message);
-    return this.create(record);
-  }
-
   /**
    * Only used for testing
    */
   async deleteExecutions() {
     const executions = await this.scan();
     return Promise.all(executions.Items.map((execution) => super.delete({ arn: execution.arn })));
   }
+
+  buildDocClientUpdateParams(item) {
+    const ExpressionAttributeNames = {};
+    const ExpressionAttributeValues = {};
+    const setUpdateExpressions = [];
+
+    Object.entries(item).forEach(([key, value]) => {
+      if (key === 'arn') return;
+      if (value === undefined) return;
+
+      ExpressionAttributeNames[`#${key}`] = key;
+      ExpressionAttributeValues[`:${key}`] = value;
+
+      if (item.status === 'running') {
+        if (['createdAt', 'updatedAt', 'timestamp', 'originalPayload'].includes(key)) {
```

Comment:
So we continuously overwrite the `originalPayload` for running executions? And for any fields besides these fields we only update them if they don't already exist? Thus, we don't update `finalPayload` or `status` in case the "completed" record had already been created

Reply:
Correct.